src/grisp_gpio.erl

```diff
@@ -220,50 +256,42 @@ pin_index(uart_2_txd)  -> 24;
 pin_index(uart_3_rxd)  -> 25;
 pin_index(uart_4_rts)  -> 26;
 pin_index(Any) when is_integer(Any), Any >= 0, Any =< 26 ->
-    Any.
-
-pin_name(0)  -> {gpio1, 1};
-pin_name(1)  -> {gpio1, 2};
-pin_name(2)  -> {gpio1, 3};
-pin_name(3)  -> {gpio1, 4};
-pin_name(4)  -> {gpio2, 1};
-pin_name(5)  -> {gpio2, 2};
-pin_name(6)  -> {gpio2, 3};
-pin_name(7)  -> {gpio2, 4};
-pin_name(8)  -> {led1, r};
-pin_name(9)  -> {led1, g};
-pin_name(10) -> {led1, b};
-pin_name(11) -> {led2, r};
-pin_name(12) -> {led2, g};
-pin_name(13) -> {led2, b};
-pin_name(14) -> {jumper, 1};
-pin_name(15) -> {jumper, 2};
-pin_name(16) -> {jumper, 3};
-pin_name(17) -> {jumper, 4};
-pin_name(18) -> {jumper, 5};
-pin_name(19) -> spi1_pin9;
-pin_name(20) -> spi1_pin10;
-pin_name(21) -> ss1;
-pin_name(22) -> ss2;
-pin_name(23) -> uart_1_cts;
-pin_name(24) -> uart_2_txd;
-pin_name(25) -> uart_3_rxd;
-pin_name(26) -> uart_4_rts.
+    Any;
+pin_index(Bad) ->
+    erlang:error(bad_arg, [pin, Bad]).
 
 map_type(periph_a) -> 0;
 map_type(periph_b) -> 1;
 map_type(periph_c) -> 2;
 map_type(periph_d) -> 3;
 map_type(input)    -> 4;
 map_type(output_0) -> 5; % default value 0
-map_type(output_1) -> 6. % default value 1
+map_type(output_1) -> 6; % default value 1
+map_type(Bad) ->
+    erlang:error(bad_arg, [type, Bad]).
```

Comment:
Use `error({bad_type, Name, Bad})` here instead of a custom error (should be `badarg`). It's easier to trace where it's coming from then (and says something more specific to the user than `badarg`).

Same goes for the other functions.

Reply:
`bad_arg` is definitely a mistake, but for the rest, I actually looked up the OTP source code and they overwhelmingly use `erlang:error(badarg, [...]).'.
e.g. https://github.com/erlang/otp/blob/master/lib/stdlib/src/dets.erl#L420

Reply:
`error/2` should get the list of arguments as input, to fill the stack trace. You're sending in an additional argument that doesn't exist.

Reply:
Yep, it seems I got the argument wrong... I should probably pass the name for better error and use `bad_type` like you suggested.

Reply:
I would prefer that. Another reason it's nice is because you know immediately where the error comes from (your own code) and don't have to hunt for the multitude of reasons a generic `badarg` could have happened.